docs/dificil/segment-trees.md

```diff
@@ -58,25 +55,25 @@ $$
 \end{aligned}
 $$
 
-!!! note "Observatie"
-    Dimensiunea $4 \cdot N$ asigura faptul ca avem suficient spatiu pentru a construi un arbore de intervale complet, indiferent de dimensiunea vectorului original. Aceasta alocare permite mentinerea unui arbore echilibrat si eficient, permitand efectuarea de operatii de actualizare si interogare intr-un mod optim. 
+!!! note "Observație"
+    Dimensiunea $4 \cdot N$ asigură faptul că avem suficient spațiu pentru a construi un arbore de intervale complet, indiferent de dimensiunea vectorului original. Această alocare permite menținerea unui arbore echilibrat și eficient, permițând efectuarea de operații de actualizare și interogare într-un mod optim.
 
-## Operatii elementare
+## Operații elementare
 
 ### Construirea arborelui
 
-Construirea unui arbore de intervale se realizeaza intr-un mod recursiv bottom-up, pornind de la frunzele arborelui (care corespund elementelor vectorului initial) si mergand spre radacina. La fiecare nivel al arborelui, valorile nodurilor se determina pe baza fiilor sai.
+Construirea unui arbore de intervale se realizează într-un mod recursiv bottom-up, pornind de la frunzele arborelui (care corespund elementelor vectorului inițial) și mergând spre rădăcină. La fiecare nivel al arborelui, valorile nodurilor se determină pe baza fiilor săi.
```

Comment:
Ar merge ca observatie sa spui undeva ca se poate construi arborele de intervale si iterativ, nu e nevoie neaparat sa insisti cu constructia lui (poate daca ajungi sa scrii si ceva mai avansat, separat de acest articol) 

Oricum am vazut ca ai dat reference la articolul asta deci il poti adauga si aici: https://codeforces.com/blog/entry/18051

Reply:
As putea adauga o nota legata de construirea lui iterativa (doar mentionata probabil), dar tind sa cred ca e mai bine sa ramana tot recursiva ca sa fie asemanatoare cu celelalte operatii si sa fie intuitiv pentru cei care dau prima data de structura asta de date. E un articol destul de introductiv, deci mai bine lasam lucrurile simple deocamdata, pentru olimpiada si concursuri e suficient parerea mea

Reply:
Da, cam asta vreau, sa fie scris undeva ca se poate si iterativ, nu trebuie sa intri in detalii. Daca articolul e introductiv, trebuie sa ai grija si la probleme sa nu fie aplicatii prea complicate.